### PR TITLE
Bug 1209025 - Fix some regressions in the revert view.

### PR DESCRIPTION
**Pull request by jezdez**
Also does:
- add some code comments for easier understanding
- handle database transaction inside model method, not view
- refresh model instance before making revision the current one by resetting ORM object with related info

Seems like a regression from #3483

### CODE REVIEW

**Comment by jezdez** [comment]
@robhudson r?

File: kuma/wiki/models.py [hunk 1]
```diff
@@ -785,13 +790,17 @@ def revert(self, revision, user, comment=None):
         revision.created = datetime.now()
         revision.creator = user
         revision.save()
-        # TODO: change to primary key check instead of object comparison
```
Comment (robhudson) [inline kuma/wiki/models.py]: wonder why that was a comment when it was such a simple thing to do?

Comment (jezdez) [inline kuma/wiki/models.py]: not sure, tbh :-/

**Comment by robhudson** [comment]
The travis test run failed but the log had nothing in it so I restarted it. So far it says it is running but I see no logs again. :disappointed:

**Comment by robhudson** [comment]
The code looks good to me. I'd like to see the test failure before an official r+ but something is weird with Travis.

**Comment by jezdez** [comment]
@robhudson Ok, I've basically enabled eager mode for the test in question and then made sure the database transaction happens inside the model class helper to revert the revision correctly.

**Comment by jezdez** [comment]
@robhudson r?

File: kuma/wiki/models.py [hunk 2]
```diff
@@ -776,22 +776,46 @@ def unique_attr():
...
+        with transaction.atomic():
+
+            # reset primary key
+            revision.pk = None
+
+            # add a sensible comment
+            revision.comment = ("Revert to revision of %s by %s" %
+                                (revision.created, revision.creator))
+            if comment:
+                revision.comment += ': "%s"' % comment
```
Comment (robhudson) [inline kuma/wiki/models.py]: If `comment` has unicode, should the original string be unicode?

File: kuma/wiki/tests/test_views.py [hunk 3]
```diff
@@ -2407,6 +2405,7 @@ def _create_doc(slug, locale):
                     locale=settings.WIKI_DEFAULT_LANGUAGE))
 
     @override_config(KUMASCRIPT_TIMEOUT=1.0)
+    @override_settings(CELERY_ALWAYS_EAGER=True)
```
Comment (robhudson) [inline kuma/wiki/tests/test_views.py]: Isn't this the default when running tests?

File: kuma/wiki/views/delete.py [hunk 4]
```diff
@@ -37,9 +37,14 @@ def revert_document(request, document_path, revision_id):
...
     else:
         comment = request.POST.get('comment')
+        document = revision.document
+        old_revision_pk = revision.pk
         try:
-            with transaction.atomic():
-                revision.document.revert(revision, request.user, comment)
+            new_revision = document.revert(revision, request.user, comment)
+            # schedule a rendering of the new revision if it really was saved
+            # print new_revision.pk, revision.pk
```
Comment (robhudson) [inline kuma/wiki/views/delete.py]: Can probably remove this line

**Comment by robhudson** [comment]
r+wc

File: kuma/wiki/models.py (hunk 2 above)
Comment (jezdez) [inline kuma/wiki/models.py]: Yes.

File: kuma/wiki/tests/test_views.py (hunk 3 above)
Comment (jezdez) [inline kuma/wiki/tests/test_views.py]: Hm, is it?

File: kuma/wiki/views/delete.py (hunk 4 above)
Comment (jezdez) [inline kuma/wiki/views/delete.py]: /me flips table

File: kuma/wiki/models.py (hunk 2 above)
Comment (jezdez) [inline kuma/wiki/models.py]: For the record, this was there alrady

File: kuma/wiki/tests/test_views.py (hunk 3 above)
Comment (robhudson) [inline kuma/wiki/tests/test_views.py]: https://github.com/mozilla/kuma/blob/master/settings_test.py#L5

**Comment by jezdez** [comment]
@robhudson I removed the `CELERY_ALWAYS_EAGER` overrides as well now. want to do another pass?

**Comment by jezdez** [comment]
@robhudson r?

**Comment by robhudson** [comment]
Code looks fine. Waiting to test results.

**Comment by robhudson** [comment]
:shipit: r+

**Pull request merged**